gfx.cpp

```diff
@@ -9372,8 +9372,12 @@ class GfxInternal
         }
 
         uint64_t shader_key = 0;
-        std::string shader_key_bytecode;
-        std::string shader_key_reflection;
+        std::wstring shader_key_dir;
+        std::wstring shader_key_bytecode;
+        std::wstring shader_key_reflection;
```

Comment:
cache/pdb files are now stored using `shader_key_dir` This is built from the source code filename and path aswell as kernel entry and type. This should create a relatively unique name kernel shaders and help limit hash collisions. Files are then stored in their respective directories in a subfolder named `shader_key_dir`. With the hash key as the filename (hash keys are now converted to string as hex instead of uint).
Using a subfolder reduces hash collisions and makes it faster to search for old files as it can be done on a per kernel basis by just checking the corresponding subfolder.
Putting pdbs in a subfolder does not effect the ability for them to be used in debugging as pix will find them in subfolders automatically (tested and works fine)